From 997c52a5f05497d50ad42c5a91bb08ff03c205ae Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 9 Oct 2018 02:18:42 +0800
Subject: [PATCH 4779/5242] MLK-19827: spi: imx: add setup_wml for i.mx6ul

commit  24e19157f044e755e2771aa2753f3a049f451bd1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add new setup_wml() for i.mx6ul, otherwise, kernel crash triggered
since no such function on i.mx6ul.This issue is brought by
commit 85c124bbc1c4 ("MLK-18983-1: spi: imx: move wml setting to
later than setup_transfer").

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/spi/spi-imx.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/spi/spi-imx.c b/drivers/spi/spi-imx.c
index 433faf4..f904909 100644
--- a/drivers/spi/spi-imx.c
+++ b/drivers/spi/spi-imx.c
@@ -967,6 +967,7 @@ static void mx1_reset(struct spi_imx_data *spi_imx)
 	.trigger = mx51_ecspi_trigger,
 	.rx_available = mx51_ecspi_rx_available,
 	.reset = mx51_ecspi_reset,
+	.setup_wml = mx51_setup_wml,
 	.fifo_size = 64,
 	.has_dmamode = true,
 	.dynamic_burst = true,
@@ -1288,6 +1289,11 @@ static int spi_imx_dma_transfer(struct spi_imx_data *spi_imx,
 	if (ret)
 		return ret;
 
+	if (!spi_imx->devtype_data->setup_wml) {
+		dev_err(spi_imx->dev, "No setup_wml()?\n");
+		return -EINVAL;
+	}
+
 	spi_imx->devtype_data->setup_wml(spi_imx);
 
 	/*
-- 
1.7.9.5

